hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/Downloader.java

```diff
@@ -234,7 +245,7 @@ private Multimap<StreamFilename, StreamFileSignature> downloadAndParseSigFiles()
 
                 try {
                     var count = streamFileProvider
-                            .list(node, startAfterFilename)
+                            .list(node, startAfterFilename, getBatchSize())
```

Comment:
It's weird to pass through the batch size except that it's not the batch size by default, it's a sentinel value that indicates the provider should look up the batch size. Prefer a simpler solution that doesn't cascade to the provider classes. It's slightly less efficient since it will retrieve 200 signatures then only use one, but this code will only be called once on the very first start so doesn't need to be optimal. `getMaxSignatures()` returns Long.MAX_VALUE by default.

```suggestion
                            .list(node, startAfterFilename, getBatchSize())
                            .take(getMaxSignatures())
```